src/services/user.js

```diff
@@ -31,19 +33,45 @@ const userService = {
     }
   },
 
-  getUserById: async (id, role) => {
-    return await User.findOne({ _id: id, ...(role && { role }) })
-      .populate([
+  getUserById: async (id, role, isEdit = false) => {
+    const populateOptions = (role) => ({
+      path: `mainSubjects.${role}`,
+      populate: [
         {
-          path: 'mainSubjects.tutor',
-          select: ['-createdAt', '-updatedAt'],
-          populate: { path: 'category', select: 'name' }
+          path: 'category',
+          select: ['_id', 'name']
         },
-        { path: 'mainSubjects.student', select: ['-createdAt', '-updatedAt'] }
-      ])
+        {
+          path: 'subjects',
+          select: ['_id', 'name']
+        }
+      ]
+    })
+
+    const user = await User.findOne({ _id: id, ...(role && { role }) })
+      .populate(populateOptions('tutor'))
+      .populate(populateOptions('student'))
       .select('+lastLoginAs +isEmailConfirmed +isFirstLogin +bookmarkedOffers +videoLink')
       .lean()
       .exec()
+
+    if (!user) {
+      throw createError(404, DOCUMENT_NOT_FOUND([User.modelName]))
+    }
+
+    if (isEdit) {
+      for (const key in user.mainSubjects) {
+        const userSubjects = await Promise.all(
```

Comment:
What will happen if at least one `Promise` fails? Maybe we need to handle errors here

Reply:
Thank you! fixed, i think)